src/repositories/GatewayRepository.ts

```diff
@@ -156,18 +156,35 @@ export default class GatewayRepository extends Subject<Gateway> {
    * @returns Newly created or updated gateway object.
    */
   public async save(gateway: Gateway): Promise<Gateway> {
-    const definedOwnProps: string[] = Utils.getDefinedOwnProps(gateway);
-
-    await GatewayModel.upsert(
-      gateway,
+    const gatewayModelObj = await GatewayModel.findOne(
       {
-        fields: definedOwnProps,
+        where: {
+          gatewayAddress: gateway.gatewayAddress,
+        },
       },
     );
 
-    const updatedGateway = await this.get(
-      gateway.gatewayAddress,
-    );
+    let updatedGateway: Gateway|null;
+    if (gatewayModelObj === null) {
+      updatedGateway = this.convertToGateway(await GatewayModel.create(
+        gateway,
+      ));
+    } else {
+      const definedOwnProps: string[] = Utils.getDefinedOwnProps(gateway);
```

Comment:
getDefinedOwnProps removes column with `undefined` values. Do we need getDefinedOwnProps method since we have model validations in place now?

Reply:
Yes, we need ** getDefinedOwnProps** method.
It will help in partial update of columns for a row.